### PR TITLE
Add gasoil + thermal simulator

### PR DESCRIPTION
**Pull request by totto82**
In particular useful when simulating CO2 storage cases with thermal effects

This PR also fix a related issue with missing phases in computation of output temperature

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this update_data please

**Comment by totto82** [comment]
jenkins build this opm-tests=695 please

File: CMakeLists.txt [hunk 1]
```diff
@@ -384,7 +384,7 @@ add_dependencies(moduleVersion opmsimulators)
 set(FLOW_MODELS blackoil brine energy extbo foam gasoil gaswater
                 oilwater oilwater_brine oilwater_polymer
                 oilwater_polymer_injectivity micp polymer solvent)
-set(FLOW_VARIANT_MODELS brine_energy onephase onephase_energy)
+set(FLOW_VARIANT_MODELS brine_energy onephase onephase_energy gasoil_energy)
```
Comment (akva2) [inline CMakeLists.txt:387]: This is for variants that do not go into the main flow binary, it seems this one does so it should be in FLOW_MODELS

**Comment by totto82** [comment]
jenkins build this opm-tests=695 please

**Comment by totto82** [comment]
Can I merge this now?

**Comment by atgeirr** [comment]
> Can I merge this now?

Yes it can. I think the modification to well temperature calculation actually may fix a latent bug, in that index orders are different.

**Pull request merged**